.github/workflows/build-wheel-linux-arm64.yaml

```diff
@@ -105,26 +105,28 @@ jobs:
         path: mlir/Enzyme
 
     # Cache external project builds
-    - name: Cache LLVM Build
+    - name: Restore LLVM Build
       id: cache-llvm-build
-      uses: actions/cache@v4
+      uses: actions/cache/restore@v4
       with:
         path:  llvm-build
         key: ${{ matrix.container_name }}-llvm-${{ needs.constants.outputs.llvm_version }}-wheel-build
 
-    - name: Cache MHLO Build
+    - name: Restore MHLO Build
       id: cache-mhlo-build
-      uses: actions/cache@v4
+      uses: actions/cache/restore@v4
       with:
         path:  mhlo-build
         key: ${{ matrix.container_name }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-wheel-build
+        lookup-only: True
 
-    - name: Cache Enzyme Build
+    - name: Restore Enzyme Build
       id: cache-enzyme-build
-      uses: actions/cache@v4
+      uses: actions/cache/restore@v4
       with:
         path: enzyme-build
         key: ${{ matrix.container_name }}-enzyme-${{ needs.constants.outputs.llvm_version }}-${{ needs.constants.outputs.enzyme_version }}-wheel-build
+        lookup-only: True
```

Comment:
What does it do? And why only for Enzyme?

Reply:
It means the cache isn't actually downloaded, we only test wether there is a cache available. 

Ideally we'd do this for all caches in this section, but that would require a restructuring that I determined wasn't worth it at this time. Take the llvm build cache for example. We can't just check if the cache exists and then skip it, because if the enzyme cache doesn't exist it will need the llvm one. So we would have rearrange the order that all the caches are checked in and add additional conditions that they be downloaded even if we have a hit according to the dependency structure between them.

Instead, I only do it for Enzyme and MHLO builds because no other cache depends on them.

Reply:
Thanks :+1: